runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -401,8 +402,34 @@ static boolean isServiceEndpoint(String endpoint) {
     return Strings.isNullOrEmpty(endpoint) || Pattern.matches(ENDPOINT_REGEXP, endpoint);
   }
 
+  static void validateSdkContainerImageOptions(DataflowPipelineWorkerPoolOptions workerOptions) {
+    // Check against null - empty string value for workerHarnessContainerImage
+    // must be preserved for legacy dataflowWorkerJar to work.
+    Preconditions.checkArgument(
+        workerOptions.getWorkerHarnessContainerImage() == null
+            || workerOptions.getSdkContainerImage() == null
+            || workerOptions
+                .getWorkerHarnessContainerImage()
+                .equals(workerOptions.getSdkContainerImage()),
+        "Cannot use legacy option workerHarnessContainerImage with sdkContainerImage. Prefer sdkContainerImage.");
+
+    // Make sure the values are the same for legacy flag and current flag.
+    if (workerOptions.getSdkContainerImage() != null) {
+      workerOptions.setWorkerHarnessContainerImage(workerOptions.getSdkContainerImage());
+    } else if (workerOptions.getWorkerHarnessContainerImage() != null) {
```

Comment:
Would not this be always true since L418 set this option?

Reply:
It wouldn't go into this branch of the if condition after entering the first one.

Reply:
This shouldn't get called though if 417-418 branch is called. I do see your point on the readability of which values are null though.

Reply:
You are both absolutely right :) And thanks for improving the readability.